gst/nnstreamer/include/nnstreamer_plugin_api_single.h

```diff
@@ -0,0 +1,416 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * NNStreamer API for Tensor_Filter Sub-Plugins
+ * Copyright (C) 2021 Gichan Jang <gichan2.jang@samsung.com>
+ */
+/**
+ * @file  nnstreamer_plugin_api_single.h
+ * @date  28 Jan 2022
+ * @brief Optional/Addtional NNStreamer Single-shot APIs for sub-plugin writers. (Need Gst Devel)
+ * @see https://github.com/nnstreamer/nnstreamer
+ * @author Gichan Jang <myungjoo.ham@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+#ifndef __NNS_PLUGIN_API_SINGLE_H__
+#define __NNS_PLUGIN_API_SINGLE_H__
+
+#include <glib.h>
+#include <tensor_typedef.h>
+#include <nnstreamer_version.h>
+
+G_BEGIN_DECLS
+
+/**
+ * @brief If the given string is NULL, print "(NULL)". Copied from `GST_STR_NULL`
+ */
+
+#define _STR_NULL(str) ((str) ? (str) : "(NULL)")
+
+/**
+ * @brief Initialize the tensor info structure
+ * @param info tensor info structure to be initialized
+ */
+extern void
+gst_tensor_info_init (GstTensorInfo * info);
```

Comment:
Discussion later : let's consider not to add this header.
IMO, util functions in nnstreamer_plugin_api.h / other gst-depended function (e.g. caps-from-pad) can be moved into tensor_common.h.

Reply:
@jaeyun-jung , I changed header file name and remove some functions. Please check it again. (https://github.com/nnstreamer/nnstreamer/pull/3652/commits/a6f6077eea9e302e0b7b2f29d85d11b74ea94989)